### ISSUE TITLE
What ANLS score did you get?

### ISSUE THREAD

**Issue by SaynaEbrahimi** [herobd/layoutlmv2#1]
Thanks a lot for sharing your code! May I ask how many epochs did you train the model for and what ANLS score you got with or without using the OCRs? 

Thanks

**Comment by herobd** [comment]
The ANLS on the test set are on the leaderboards. Here's the validation ANLS using Tesseract at evaluation:
Using GT OCR during training: 2 epochs, 0.439 ANLS
                                   (retrain): 5 epochs, 0.435 ANLS
Using Tesseract during training: 2 epochs, 0.447 ANLS

**Comment by SaynaEbrahimi** [comment]
Thank you for your quick reply! 

Have you ever trained it for long enough to get to the results of the LayoutLMv2 paper (0.7808)? Thanks to your code, I was able to get 0.651 after one epoch  and 0.7392 after 12 epochs with GT OCR on the validation set. I used total batch size of 64 (8 GPUS, batch size of 8 each using DistributedDataParallel training). So we can get a score on the test set if we submit to the leaderboard? Is it open for submissions?

**Comment by herobd** [comment]
I had LayoutLMv2 overfitting after 2 epochs, this may have been because I
was using a batch size of 5. Maybe I'll drop the LR or something.
There's also going to be a performance difference because I'm using the
Tesseract OCR for evaluation (even when training with GT OCR). My
experiment was explicitly to show the choice of OCR can be very important.

Ya, anyone can submit results and get a score.

**Comment by SaynaEbrahimi** [comment]
I see. Thanks for the information and again for your code!

**Comment by haolibai** [comment]
> Thank you for your quick reply!
> 
> Have you ever trained it for long enough to get to the results of the LayoutLMv2 paper (0.7808)? Thanks to your code, I was able to get 0.651 after one epoch and 0.7392 after 12 epochs with GT OCR on the validation set. I used total batch size of 64 (8 GPUS, batch size of 8 each using DistributedDataParallel training). So we can get a score on the test set if we submit to the leaderboard? Is it open for submissions?

Hi, may I know your detailed experiemntal settings to produce 0.7392 ANLS score? (e.g., base model or large model? learning rates and epochs.) I am also reproducing DocVQA but can only get around 0.65~ ANLS score with the base model. It seems the task is quite tricky, both for the pre-processing and post-processing.

**Comment by SaynaEbrahimi** [comment]
@haolibai  Hi, let me emphasize again that I got 0.7392 on the validation set (not test set which is the data used for the results on their leaderboard). I used the base model which has ~200M parameters (`layoutlmv2-base-uncased`) with learning rate of 5e-5 and using the original OCRs. I mentioned my batch size above. I think the results shown in LayoutLMv2 paper on DocVQA with the base model is on the actual test set and not the validation set and I agree with you that getting that high number is tricky. I haven't been able to achieve that. I was only able to get ~0.50 on the test set with the same model. What are your scores on the validation and test sets? I am happy to share thoughts to find the tricks behind this mysterious ANLS score!
